src/server/artifact.ts

```diff
@@ -16,24 +16,27 @@
 
 import fs from 'fs';
 import * as util from 'util';
+import { Download } from './download';
 import { SdkObject } from './instrumentation';
 
 type SaveCallback = (localPath: string, error?: string) => Promise<void>;
 
 export class Artifact extends SdkObject {
   private _localPath: string;
   private _unaccessibleErrorMessage: string | undefined;
+  private _owner: Download | undefined;
   private _finishedCallback: () => void;
   private _finishedPromise: Promise<void>;
   private _saveCallbacks: SaveCallback[] = [];
   private _finished: boolean = false;
   private _deleted = false;
   private _failureError: string | null = null;
 
-  constructor(parent: SdkObject, localPath: string, unaccessibleErrorMessage?: string) {
+  constructor(parent: SdkObject, localPath: string, unaccessibleErrorMessage?: string, owner?: Download) {
     super(parent, 'artifact');
     this._localPath = localPath;
     this._unaccessibleErrorMessage = unaccessibleErrorMessage;
+    this._owner = owner;
```

Comment:
Let's pass `CancellCallback` function instead to limit the information we expose to the Artifact.